transform/snowflake-dbt/models/blapi/customers_with_onprem_subs.sql

```diff
@@ -24,6 +24,24 @@ WITH latest_credit_card_address AS (
         ON addresses.country = postal_code_mapping.country
         AND postal_code_mapping.postal_code like addresses.postal_code || '%'
     WHERE addresses.address_type = 'billing'
+), latest_credit_card_address AS (
+    -- Country from blapi addresses is inconsistent. Sometimes it's country name, while others it's country code.
+    -- This CTE attempts to normalize this.
+    SELECT
+        pa.customer_id,
+        pa.line1,
+        pa.line2,
+        pa.postal_code,
+        pa.city,
+        pa.state,
+        COALESCE(cc.name, pa.country) AS country,
+        pa.state_code,
+        COALESCE(pa.country_code, cc.code) AS country_code
+    FROM
+        payment_addresses pa
+        LEFT JOIN {{ ref('country_codes') }} cc ON pa.country = cc.code
```

Comment:
`pa.country = cc.code` -> `pa.country_code = cc.code` ?

Reply:
> `pa.country = cc.code` -> `pa.country_code = cc.code` ?

It's practically the same due to the `COALESCE` exception. `addresses` seem to contain country codes in `country` field. Whenever country code or country from postal code mapping is null, the country code is used (for both!). However updating as it's a bit more readable.

A side note is that CTE `latest_credit_card_address` is used in other models. It can also be simplified and become more consistent. Moving it to a separate model can help avoid repeating code, keep data consistent and simplify things. However this is quite some effort.

Reply:
"addresses seem to contain country codes in country field" - think we should open a bug on BLAPI/Nick team to address this long-term (they might improve other validations as well while they are in the area)